File: src/main/java/me/viniciusarnhold/altaria/command/interfaces/ICommand.java
package me.viniciusarnhold.altaria.command.interfaces;

import me.viniciusarnhold.altaria.command.CommandType;
import me.viniciusarnhold.altaria.command.UserPermissions;
import org.jetbrains.annotations.NotNull;

import java.util.EnumSet;
import java.util.Set;

/**
 * Created by Vinicius.
 *
 * @since ${PROJECT_VERSION}
 */
public interface ICommand {

    @NotNull
    String command();

    @NotNull
    Set<String> aliases();

    @NotNull
    String description();

    @NotNull
    CommandType type();

    @NotNull
    EnumSet<UserPermissions> permissions();


}

File: core/src/main/kotlin/me/viniciusarnhold/altaria/command/interfaces/ICommand.kt
package me.viniciusarnhold.altaria.command.interfaces

import me.viniciusarnhold.altaria.command.CommandType
import me.viniciusarnhold.altaria.command.UserPermissions
import java.util.*

/**
 * Created by Vinicius.

 * @since ${PROJECT_VERSION}
 */
interface ICommand {

    fun command(): String

    fun aliases(): Set<String>

    fun description(): String

    fun type(): CommandType

    fun permissions(): EnumSet<UserPermissions>


}
